### PR TITLE
Update Webpack config and dependencies

### PR DESCRIPTION
**Pull request by abukurov**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,10 +7,11 @@
     "lint": "eslint frontend/** && stylelint \"frontend/**/*.css\"",
     "test": "npm run lint",
     "stylefmt": "stylefmt -r frontend/blocks/**/*.css",
-    "build": "NODE_ENV=production webpack",
-    "watch": "webpack",
-    "start": "npm run watch & node app.js",
     "precommit": "npm test"
```
Comment (102) [inline package.json]: missing comma

File: webpack.dev.config.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      })
+    }]
+  },
+
+  plugins: [
+    new webpack.HotModuleReplacementPlugin(),
+    new ExtractTextPlugin('[name].css')
+  ]
+};
+
```
Comment (102) [inline webpack.dev.config.js]: extra newline

File: examples/index.html [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        <th>Провайдер</th>
+        <th></th>
+      </tr>
+      </thead>
+
+      <tbody></tbody>
+    </table>
+  </div>
+  <!-- Latest compiled and minified JavaScript -->
+  <script src="//code.jquery.com/jquery-2.2.4.min.js" integrity="sha256-BbhdlvQf/xTY9gja0Dq3HiwQF8LaCRTXxZKRutelT44=" crossorigin="anonymous"></script>
```
Comment (maxmalov) [inline examples/index.html]: should be listed in dev dependencies, CDN can be unavailable

File: examples/index.html [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  <script src="//maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
```
Comment (maxmalov) [inline examples/index.html]: same as above

File: webpack.config.js [hunk 5]
```diff
@@ -1,92 +1,69 @@
...
-const NODE_ENV = process.env.NODE_ENV || 'development';
-let ExtractTextPlugin = require('extract-text-webpack-plugin');
+const webpack = require('webpack');
+const CompressionPlugin = require('compression-webpack-plugin');
+const ExtractTextPlugin = require('extract-text-webpack-plugin');
+const LodashModuleReplacementPlugin = require('lodash-webpack-plugin');
 
 module.exports = {
-  context: path.join(__dirname, 'frontend'),
+  devtool: 'cheap-source-map',
```
Comment (maxmalov) [inline webpack.config.js]: Should be `source-map` for production builds

File: webpack.config.js [hunk 6]
```diff
@@ -1,92 +1,69 @@
...
-    "jquery": "jQuery"
+    jquery: 'jQuery'
   },
 
-  watch: NODE_ENV === 'development',
-
-  watchOptions: {
-    aggregateTimeout: 100
+  module: {
+    rules: [{
```
Comment (maxmalov) [inline webpack.config.js]: Copy pasted from dev configuration :scream:

File: webpack.common.config.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
+/* eslint-env node */
+
+const path = require('path');
+const ExtractTextPlugin = require('extract-text-webpack-plugin');
+
+module.exports = {
+
+
```
Comment (maxmalov) [inline webpack.common.config.js]: too many blank lines

File: webpack.config.js [hunk 8]
```diff
@@ -1,92 +1,33 @@
...
+const LodashModuleReplacementPlugin = require('lodash-webpack-plugin');
 
-  watchOptions: {
-    aggregateTimeout: 100
-  },
+module.exports = new WebpackConfig().extend('./webpack.common.config.js').merge({
+  devtool: 'source-map',
 
-  devtool: (NODE_ENV === 'development') ? 'cheap-inline-module-source-map' : null,
+  entry: path.join(__dirname, 'frontend', 'index.js'),
```
Comment (maxmalov) [inline webpack.config.js]: you can use `require.resolve` for `.js` files

**Pull request merged**